pkg/domain/repositories/cluster_access_repo.go

```diff
@@ -122,13 +143,35 @@ func (r *clusterAccessRepository) GetClustersAccessibleByUserId(ctx context.Cont
 
 			// get accessible cluster by tenant and append
 			var bindings []*domain_projections.TenantClusterBinding
-			bindings, err = r.tenantClusterBindingRepo.GetByTenantId(ctx, uuid.MustParse(binding.GetResource()))
+			bindings, err = r.tenantClusterBindingRepo.GetByTenantId(ctx, tenantId)
 			if err != nil {
 				return
 			}
-			clusters, err = r.getClustersByBindings(ctx, bindings, k8sRoles)
```

Comment:
Here was the actual problem that people got flaky clusters available to them. The return variable was always overwritten and the different sources of access where not merged into one result.